antsibull/app_context.py

```diff
@@ -215,20 +216,39 @@ class AppContext(BaseModel):
         values stored in extras need default values, they need to be set outside of the context
         or the entries can be given an actual entry in the AppContext to take advantage of the
         schema's checking, normalization, and default setting.
+    :ivar ansible_base_url: Url to the ansible-core git repo.
+    :ivar breadcrumbs: If True, build with breadcrumbs on the plugin pages (this takes more memory).
     :ivar galaxy_url: URL of the galaxy server to get collection info from
+    :ivar indexes: If True, create index pages for all collections and all plugins in a collection.
     :ivar logging_cfg: Configuration of the application logging
-    :ivar pypi_url: URL of thepypi server to query for information
+    :ivar pypi_url: URL of the pypi server to query for information
     """
 
     extra: ContextDict = ContextDict()
     # pyre-ignore[8]: https://github.com/samuelcolvin/pydantic/issues/1684
     ansible_base_url: p.HttpUrl = 'https://github.com/ansible/ansible/'
+    breadcrumbs: p.StrictBool = True
     # pyre-ignore[8]: https://github.com/samuelcolvin/pydantic/issues/1684
     galaxy_url: p.HttpUrl = 'https://galaxy.ansible.com/'
+    indexes: p.StrictBool = True
     logging_cfg: LoggingModel = LoggingModel.parse_obj(DEFAULT_LOGGING_CONFIG)
     # pyre-ignore[8]: https://github.com/samuelcolvin/pydantic/issues/1684
     pypi_url: p.HttpUrl = 'https://pypi.org/'
 
+    @p.validator('breadcrumbs', 'indexes', pre=True)
+    def convert_to_bool(cls, value):
+        if isinstance(value, str):
+            if value.lower() in ('0', 'false', 'no', 'n', 'f', ''):
+                value = False
+            else:
```

Comment:
I would have an explicit list of values here, and fail if the string isn't one of these.

This avoids users who had a typo (`fals` instead of `false`) to accidentally enable the feature.

Reply:
Fixed in 7d717282a48792c671de1963fb204e868c613ec9

https://github.com/ansible-community/antsibull/pull/292/commits/7d717282a48792c671de1963fb204e868c613ec9#diff-02cd3e7ff7dcd7b35c8080c275011994c07d608c3e9be75336848184f1c498b1R25

I'm a little unhappy with the implementation of int values and strings which can be int but I think the logic is right.  0 and string representations of that (ex: `"0"`) are False and any other int is True.  (The implementation isn't trivial because booleans are int subclasses in python.  I wanted to avoid converting them twice so I ended up leaving `isinstance(int)` in an `ifelse` after checking for str and then calling `_is_truthy_int()` in two separate code paths.